### ISSUE TITLE
metaSPADESOOM errors on small datasets

### ISSUE THREAD

**Issue by Lamm-a** [ablab/spades#990]
### Description of bug

I am running metaSPADES on a co-assembly of 6 samples totalling 19gb raw reads given to metaSPADES. Despite this being a somewhat typical size of metagenomic dataset I get out of memory errors when giving 170 GB of memory (and 16 threads). I know this is a me problem but I am just unsure how to process as that is a huge amount of memory already and it seems crazy that it OOMs.

Do you have any advice on how I could proceed? metaSPADES provides better assemblies that other tools out there so I would like to use it.

I understand there are other factors but is it possible for you to provide usage data similar to how you do for spades in the README?

### spades.log

[spades.log](https://github.com/ablab/spades/files/8910743/spades.log)

### params.txt

[params.txt](https://github.com/ablab/spades/files/8910745/params.txt)

### SPAdes version

3.14.0

### Operating System

CentOS Linux release 7.6.1810 (Core)

### Method of SPAdes installation

Manual

### No errors reported in spades.log

- [X] Yes

**Comment by Lamm-a** [comment]
The error still occurs on the latest version of SPAdes

**Comment by asl** [comment]
@Lamm-a will you please post new spades.log

**Comment by asl** [comment]
Yes, you definitely need more RAM for such large datasets. You're having ~15 billion k-mers there.

**Comment by Lamm-a** [comment]
Any predictions on how much?

Will skipping k-mer 21 drastically reduce the quality of the assembly? As that will reduce the k-mer count by a lot.

**Comment by asl** [comment]
There is no way to predict the memory usage before the assembly as it depends on many things including the genome size, repeat content, error rate, strain content, etc.

Skipping shorter k-mers will not reduce the k-mer count in any way. Quite the opposite.

**Comment by Lamm-a** [comment]
> Skipping shorter k-mers will not reduce the k-mer count in any way. Quite the opposite.

Oh how come that is the case?

Would you suggest now merging these into a co assembly then? As it looks like the amount of memory I will need will be prohibitive for me.

**Comment by asl** [comment]
> Oh how come that is the case?

Pretty simple. We are talking about the # of distinct k-mers here. Also observe that each sequencing error introduces `k` non-genomic k-mers.

> Would you suggest now merging these into a co assembly then? As it looks like the amount of memory I will need will be prohibitive for me.

What do you want to merge? You're having just a single sample here. Also, merging samples and co-assembling required much more RAM as you'd assemble all data at once.

**Comment by Lamm-a** [comment]
> What do you want to merge? You're having just a single sample here. Also, merging samples and co-assembling required much more RAM as you'd assemble all data at once.

Sorry that was a typo, I meant *not* merging.

**Comment by asl** [comment]
Well, you could certainly try to merge less and see what will be the outcome

**Comment by Lamm-a** [comment]
> Pretty simple. We are talking about the # of distinct k-mers here. Also observe that each sequencing error introduces k non-genomic k-mers.

Ah I don't think I understand what is going on with the k-mer stuff then. I understood it as each k-mer length would generate its own set of mers.

**Comment by zfsh** [comment]
Hello, I would like to know if you solved this problem, how did you solve it. Because I'm having the same problem at the moment, I hope to get some advice from you，thanks.

**Comment by gnasseruncc** [comment]
I am facing the same issue at the same point as the second spades log you provided. I have been trying to get past it but I have not yet, did a solution come up for this? Thanks!

**Comment by allierobertsmoore** [comment]
Also facing this issue. Huge amount of memory being consumed (>500Gb). Is there any way around this aside from ^^ RAM  server?
